internal/driver/drv_uart_generic.go

```diff
@@ -83,7 +86,11 @@ func (dev *UartGeneric) GenericUartRead(maxbytes int) error {
 		lens, err := dev.conn.Read(b)
 
 		if err != nil {
-			log.Printf("GenericUartRead(): Error = %v", err)
+			if err == io.EOF {
+				log.Printf("GenericUartRead(): %v - Finished reading!", err)
+				break
+			}
+			log.Printf("GenericUartRead(): Exit - Error = %v", err)
```

Comment:
Please add a new parameter with `logger.LoggingClient` type to `GenericUartRead` function, and pass `s.lc` (loggingClient) from the parent function to use `LoggingClient` instead of the standard `log` library, to have a consistent log format.

Reply:
Addressed.